deeppavlov/models/go_bot/trippy_preprocessing.py

```diff
@@ -0,0 +1,970 @@
+
+    Adapted from TripPy get_tok_label.
+    The position of the last occurrence of the slot value will be used.
+    """
+    sys_utt_tok_label = [0 for _ in sys_utt_tok]
+    usr_utt_tok_label = [0 for _ in usr_utt_tok]
+    if slot_type not in cur_ds_dict:
+        class_type = 'none'
+    else:
+        value = cur_ds_dict[slot_type]
+        if value == 'dontcare' and (slot_type not in prev_ds_dict or prev_ds_dict[slot_type] != 'dontcare'):
+            # Only label dontcare at its first occurrence in the dialog
+            class_type = 'dontcare'
+        else: # If not none or dontcare, we have to identify whether
+            # there is a span, or if the value is informed
+            in_usr = False
+            in_sys = False
+            for label_d in usr_slot_label:
+                if (label_d['slot'] == slot_type) and (label_d['start'] >= 0):
+                    # Allow start & end logits referring to utterances that do not exactly match the slot value; Not in original TripPy
+                    # Start should still be >0 not to allow ones with -1, i.e. there is not value
+                    #value == ' '.join(usr_utt_tok[label_d['start']:label_d['exclusive_end']]):
+                    for idx in range(label_d['start'], label_d['exclusive_end']):
+                        usr_utt_tok_label[idx] = 1
+                    in_usr = True
+                    class_type = 'copy_value'
+                    if slot_last_occurrence:
+                        break
+
+            # This is never used for DSTC2, as there are no sys_slot_labels, however, we leave it for possible future use & TripPy conformity
+            for label_d in sys_slot_label:
+                if label_d['slot'] == slot_type and value == ' '.join(
+                        sys_utt_tok[label_d['start']:label_d['exclusive_end']]):
+                    for idx in range(label_d['start'], label_d['exclusive_end']):
+                        sys_utt_tok_label[idx] = 1
+                    in_sys = True
+                    if not in_usr or not slot_last_occurrence:
+                        class_type = 'inform'
+                    if slot_last_occurrence:
+                        break
+
+            if not in_usr and not in_sys:
+                assert sum(usr_utt_tok_label + sys_utt_tok_label) == 0
+                if (slot_type not in prev_ds_dict or value != prev_ds_dict[slot_type]):
+                    # Added clarifications
+                    print("Value: {} for slot type {} does not exist in utterance: {}".format(value, slot_type, usr_utt_tok))
+                    print("Other Information: ", prev_ds_dict, usr_slot_label, sys_utt_tok)
+                    print("Most likely have to add the incorrect utterance token to be replaced with the value see normalizations in the code.")
+                    print("This error is likely because of a mismatch in the slot value & the actual text!\n \
```

Comment:
either `log` this or include in the error message

Reply:
Now logged!